hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -272,6 +275,28 @@ public void notifyCheckpointComplete(long checkpointId) {
     );
   }
 
+  private void emitCompactionAndClusteringMetrics(Configuration conf,
+      HoodieTableMetaClient metaClient, HoodieFlinkWriteClient writeClient) {
+    if (conf.getBoolean(FlinkOptions.CLUSTERING_SCHEDULE_ENABLED)
+        && !conf.getBoolean(FlinkOptions.CLUSTERING_ASYNC_ENABLED)) {
+      HoodieTimeline pendingReplaceTimeline = metaClient.getActiveTimeline()
+          .filterPendingReplaceTimeline();
+      HoodieMetrics metrics = writeClient.getMetrics();
+      if (metrics != null) {
+        metrics.setPendingClusteringCount(pendingReplaceTimeline.countInstants());
+      }
+    }
+    if (conf.getBoolean(FlinkOptions.COMPACTION_SCHEDULE_ENABLED)
```

Comment:
We already have the pending compaction instants metrics support in https://github.com/apache/hudi/pull/9515, does that work for you?

Reply:
@danny0405 en，i get you, however, this is another situation, when we FlinkOptions.COMPACTION_SCHEDULE_ENABLED set to true, FlinkOptions.COMPACTION_ASYNC_ENABLED set to false ， the CompactionPlanOperator will not be added to pipeline, then will not report the pending compaction plan , the situation is as same as clustering. this situation happens to offload the clustering or compaction.

Reply:
yeah, you are right, there are two set of metrics for Flink now, we might need to unify them.

Reply:
And we may need to reload the active timeline to refresh the instant state.

Reply:
> And we may need to reload the active timeline to refresh the instant state.

ok, i'll fit .